### PR TITLE
SONAR-8857 api/qualityprofiles/projects must filter on org

### PR DESCRIPTION
**Pull request by bartfastiel**
When searching for not yet associated projects of a quality profile, only projects of the same organization may be returned.

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java [hunk 1]
```diff
@@ -73,9 +73,9 @@
...
   void deleteProjectAssociationByProfileKeys(@Param("profileKeys") Collection<String> profileKeys);
 
-  List<ProjectQprofileAssociationDto> selectSelectedProjects(@Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
+  List<ProjectQprofileAssociationDto> selectSelectedProjects(@Param("organizationUuid") String organizationUuid, @Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
 
-  List<ProjectQprofileAssociationDto> selectDeselectedProjects(@Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
+  List<ProjectQprofileAssociationDto> selectDeselectedProjects(@Param("organizationUuid") String organizationUuid, @Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
 
-  List<ProjectQprofileAssociationDto> selectProjectAssociations(@Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
+  List<ProjectQprofileAssociationDto> selectProjectAssociations(@Param("organizationUuid") String organizationUuid, @Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java]: MINOR Split this 194 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java [hunk 2]
```diff
@@ -73,9 +73,9 @@
 
   void deleteProjectAssociationByProfileKeys(@Param("profileKeys") Collection<String> profileKeys);
 
-  List<ProjectQprofileAssociationDto> selectSelectedProjects(@Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
+  List<ProjectQprofileAssociationDto> selectSelectedProjects(@Param("organizationUuid") String organizationUuid, @Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java]: MINOR Split this 191 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java [hunk 3]
```diff
@@ -73,9 +73,9 @@
...
 
-  List<ProjectQprofileAssociationDto> selectDeselectedProjects(@Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
+  List<ProjectQprofileAssociationDto> selectDeselectedProjects(@Param("organizationUuid") String organizationUuid, @Param("profileKey") String profileKey, @Param("nameQuery") String nameQuery);
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileMapper.java]: MINOR Split this 193 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml [hunk 4]
```diff
@@ -137,36 +137,33 @@
...
     JOIN project_qprofiles pp ON pp.project_uuid = pj.uuid
-    AND pp.profile_key = #{profileKey}
-    <where>
-      AND pj.scope='PRJ' AND pj.qualifier='TRK'
-      AND UPPER(pj.name) LIKE #{nameQuery}
-    </where>
+    AND pp.profile_key = #{profileKey, jdbcType=VARCHAR}
+    WHERE pj.scope='PRJ' AND pj.qualifier='TRK'
+      AND UPPER(pj.name) LIKE #{nameQuery, jdbcType=VARCHAR}
+      AND pj.organization_uuid LIKE #{organizationUuid, jdbcType=VARCHAR}
```
Comment (simonbrandhof) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml]: not `like` but `=`

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml [hunk 5]
```diff
@@ -137,36 +137,33 @@
...
-    AND pp.profile_key = #{profileKey}
-    <where>
-      AND pj.scope='PRJ' AND pj.qualifier='TRK'
-      AND UPPER(pj.name) LIKE #{nameQuery}
+    AND pp.profile_key = #{profileKey, jdbcType=VARCHAR}
+    WHERE pj.scope='PRJ' AND pj.qualifier='TRK'
+      AND UPPER(pj.name) LIKE #{nameQuery, jdbcType=VARCHAR}
       AND pp.profile_key IS NULL
-    </where>
+      AND pj.organization_uuid LIKE #{organizationUuid, jdbcType=VARCHAR}
```
Comment (simonbrandhof) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml]: not like but =

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml [hunk 6]
```diff
@@ -137,36 +137,33 @@
...
     LEFT JOIN project_qprofiles pp ON pp.project_uuid = pj.uuid
-    AND pp.profile_key = #{profileKey}
-    <where>
-      AND pj.scope='PRJ' AND pj.qualifier='TRK'
-      AND UPPER(pj.name) LIKE #{nameQuery}
-    </where>
+    AND pp.profile_key = #{profileKey, jdbcType=VARCHAR}
+    WHERE pj.scope='PRJ' AND pj.qualifier='TRK'
+      AND UPPER(pj.name) LIKE #{nameQuery, jdbcType=VARCHAR}
+      AND pj.organization_uuid LIKE #{organizationUuid, jdbcType=VARCHAR}
```
Comment (simonbrandhof) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/QualityProfileMapper.xml]: not like but =

**Pull request merged**